mongoWellnessChecker/mongoWellnessChecker.js

```diff
@@ -0,0 +1,592 @@
+ * or data loss incurred with their use.
+ *
+ * You are responsible for reviewing and testing any scripts you run
+ * thoroughly before use in any non-testing environment.
+ *
+ *
+ * LICENSE
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+//Make the ObjectID format in the output same as the legacy getMongoData script
+if (typeof (ObjectId) != "undefined") {
+
+	ObjectId.prototype.toJSON = ObjectId.prototype.toExtendedJSON;
+}
+
+//Make the Timestamp format in the output same as the legacy getMongoData script
+if (typeof (Timestamp) != "undefined") {
+
+
+    Timestamp.prototype.toJSON = function() {
+        return this.toStringIncomparable(); 
+     };
+
+    Timestamp.prototype.toStringIncomparable = function() {
+        var t = this.hasOwnProperty("t") ? this.t : this.high;
+        var i = this.hasOwnProperty("i") ? this.i : this.low;
+	var timestamplegacy = {};
+        var tsdoc = {} ;
+        tsdoc['t'] = t;
+        tsdoc['i'] = i;
+	timestamplegacy['$timestamp']=tsdoc;
+        return timestamplegacy;
+
+    };
+} else {
+    print("warning: no Timestamp class");
+}
+
+// Taken from legacy getMongoData.js script and then modified it to work with mongosh 
+if (typeof print.captureAllOutput === "undefined") {
+    print.captureAllOutput = function (fn, args) {
+        var res = {};
+        res.output = [];
+        var __orig_print = print;
+        print = function () { 
+		Array.prototype.push.apply(res.output, Array.prototype.slice.call(arguments).join(" ").split("\n")); 
+	};
+        try { 
+            res.result = fn.apply(undefined, args);
+        } catch(e) { 
+		print(e);
+	}
+        finally {
+            // Stop capturing print() output
+            print = __orig_print;
+        }
+        return res;
+    };
+}
+
+// For use in JSON.stringify to properly serialize known types
+function jsonStringifyReplacer(k, v){
+    //if (v instanceof ObjectId)
+    //    return { "$oid" : v.toString() };
+    if (v instanceof Long)
+        return { "$numberLong" : longmangle(v) };
+    if (v instanceof Int32)
+        return v.toNumber();
+    // For ISODates; the $ check prevents recursion
+    if (typeof v === "string" && k.startsWith('$') == false){
+        try {
+            iso = ISODate(v);
+            return { "$date" : iso.valueOf() };
+        }
+        // Nothing to do here, we'll get the return at the end
+        catch(e) {}
+    }
+    return v;
+}
```

Comment:
I'd consider dropping this function and using `EJSON.stringify(_output, null, 4);` instead of `JSON.stringify(_output, jsonStringifyReplacer, 4` in the last line, that handles all relevant BSON types' conversion to EJSON

Reply:
ack

Reply:
There are legacy analyzer programs that only work with legacy getMongoData.js output. Using EJSON.stringify could cause issues with those. Hence, we are sticking to JSON.stringify for now.